libsignal-service-actix/src/push_service.rs

```diff
@@ -268,20 +271,24 @@ impl PushService for AwcPushService {
         &mut self,
         endpoint: Endpoint,
         path: &str,
+        additional_headers: &[(&str, &str)],
         credentials_override: HttpAuthOverride,
         value: S,
     ) -> Result<D, ServiceError>
     where
         for<'de> D: Deserialize<'de>,
         S: Serialize,
     {
-        let mut response = self
-            .request(Method::PUT, endpoint, path, credentials_override)?
-            .send_json(&value)
-            .await
-            .map_err(|e| ServiceError::SendError {
+        let mut request =
+            self.request(Method::PUT, endpoint, path, credentials_override)?;
+        for &header in additional_headers {
```

Comment:
how about adding an additional parameters for `additional_headers` to the `self.request()` method instead?

Reply:
... because in my head somehow I read `client.request`, didn't notice it was a method we controlled and could change the signature of. Good catch!